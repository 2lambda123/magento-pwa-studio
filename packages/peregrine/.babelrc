{
  "plugins": [
    "transform-object-rest-spread",
    "transform-class-properties",
    ["transform-react-jsx", { "pragma": "createElement" }],
    ["transform-runtime", {
      "helpers": true,
      "polyfill": false, // polyfills will be handled by preset-env
      "regenerator": false
    }]
  ],
  "presets": [
    ["env", {
      "targets": {
        "browsers": ["> 5%"]
      },
      "modules": false
    }]
  ],
  "env": {
    "test": {
      "plugins": [
        "transform-es2015-modules-commonjs",
        "transform-object-rest-spread",
        "transform-class-properties",
        ["transform-react-jsx", { "pragma": "createElement" }]
      ]
    }
  }
}
