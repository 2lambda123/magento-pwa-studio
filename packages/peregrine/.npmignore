.eslintrc.js
.storybook
/lib/**/__{docs,helpers,mocks,stories,tests}__/**
/scripts
