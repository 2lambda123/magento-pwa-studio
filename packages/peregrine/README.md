# Peregrine

Peregrine is a collection of UI components for Magento PWA projects.

## Under construction

This repository is currently under reconstruction and is not suitable for use outside of the Venia storefront. Stay tuned for more information on the vision for `peregrine`.

## Contributing

The `peregrine` repository is an open source project that welcomes contributors of all skill levels.

If you want to contribute to this project, please review the [contribution guidelines] and follow our [code of conduct].

## License

This project is under the OSL-3.0 license - see the [LICENSE] file for details.

[Peregrine documentation]: https://magento-research.github.io/pwa-studio/peregrine/
[contribution guidelines]: .github/CONTRIBUTING.md
[code of conduct]: .github/CODE_OF_CONDUCT.md
[LICENSE]: LICENSE
