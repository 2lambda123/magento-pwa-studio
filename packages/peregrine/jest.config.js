module.exports = {
    displayName: 'Peregrine',
    setupFiles: [
        '<rootDir>/scripts/shim.js',
        '<rootDir>/scripts/fetch-mock.js'
    ],
    browser: true,
    testURL: 'https://localhost/',
    coveragePathIgnorePatterns: ['scripts/*', 'node_modules', 'src/index.js'],
    testPathIgnorePatterns: ['dist', 'node_modules']
};
