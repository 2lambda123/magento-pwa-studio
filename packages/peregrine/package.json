{
  "name": "@magento/peregrine",
  "version": "3.0.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "The core runtime of Magento PWA",
  "main": "lib/index.js",
  "scripts": {
    "clean": " ",
    "storybook": "start-storybook -p 9001 -c .storybook",
    "storybook:build": "build-storybook -c .storybook -o storybook-dist"
  },
  "repository": "github:magento-research/pwa-studio",
  "author": "Magento Commerce",
  "license": "(OSL-3.0 OR AFL-3.0)",
  "bugs": {
    "url": "https://github.com/magento-research/pwa-studio/issues"
  },
  "homepage": "https://github.com/magento-research/pwa-studio/tree/master/packages/peregrine#readme",
  "devDependencies": {
    "@storybook/react": "~4.1.13",
    "intl": "~1.2.5",
    "react": "~16.8.3",
    "react-apollo": "~2.5.3",
    "react-dom": "~16.8.3",
    "react-router-dom": "~5.0.0",
    "react-test-renderer": "~16.8.6"
  },
  "peerDependencies": {
    "@babel/runtime": "^7.4.2",
    "react": "^16.8.0",
    "react-apollo": "^2.5.3",
    "react-dom": "^16.8.0",
    "react-router-dom": "~5.0.0"
  },
  "engines": {
    "node": "10.x",
    "yarn": ">=1.12.0"
  },
  "module": "lib/index.js",
  "jsnext:main": "lib/index.js",
  "es2015": "lib/index.js",
  "sideEffects": false
}
