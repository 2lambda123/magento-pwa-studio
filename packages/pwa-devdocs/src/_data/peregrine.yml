title: Peregrine
entries:
  - label: Overview
    url: /peregrine/

  - label: Storybook UI testing
    url: /peregrine/testing/storybook/

  - label: Utility modules
    entries:

      - label: REST API client
        url: /peregrine/reference/rest-api-client/

  - label: React components
    entries:

      - label: ContainerChild
        url: /peregrine/reference/container-child/

      - label: Item
        url: /peregrine/reference/item/

      - label: Items
        url: /peregrine/reference/items/

      - label: List
        url: /peregrine/reference/list/

      - label: Router
        url: /peregrine/reference/router/
