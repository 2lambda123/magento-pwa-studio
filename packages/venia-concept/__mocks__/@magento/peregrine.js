import React from 'react';
import { BrowserPersistence } from './util/simplePersistence';

const peregrine = jest.requireActual('../../../peregrine/src');

// re-exports
const RestApi = {
    Magento2: {
        request: jest.fn()
    }
};

const Util = { BrowserPersistence };

// hooks
const useApolloContext = jest.fn(peregrine.useApolloContext);
const useDocumentListener = jest.fn(peregrine.useDocumentListener);
const useDropdown = jest.fn(peregrine.useDropdown);
const useQuery = jest.fn(peregrine.useQuery);
const useQueryResult = jest.fn(peregrine.useQueryResult);
const useSearchParam = jest.fn(peregrine.useSearchParam);

// components

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
const Price = jest.fn().mockReturnValue(<div />);

module.exports = {
    ...peregrine,
    Price,
    RestApi,
    Util,
    useApolloContext,
    useDocumentListener,
    useDropdown,
    useQuery,
    useQueryResult,
    useSearchParam
};
