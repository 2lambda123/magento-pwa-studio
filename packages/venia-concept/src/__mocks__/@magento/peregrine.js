import { createElement } from 'react';

/**
 * the Price component from @magento/peregrine
 * has browser-specific functionality and cannot
 * currently by rendered in the test environment
 */
export const Price = () => <div />;
