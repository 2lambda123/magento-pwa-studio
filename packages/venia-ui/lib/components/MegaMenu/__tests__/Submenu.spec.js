import React from 'react';
import { createTestInstance } from '@magento/peregrine';
import { MemoryRouter } from 'react-router-dom';

import Submenu from '../submenu';

jest.mock('../../../classify');

describe('Submenu renders correctly', () => {
    const props = {
        items: [
            {
                id: 1,
                name: 'Bottoms',
                url_path: 'bottoms',
                url_suffix: '.html',
                children: [
                    {
                        id: 3,
                        name: 'Pants',
                        url_path: 'pants',
                        url_suffix: '.html',
                        isActive: false,
                        children: []
                    }
                ]
            },
            {
                id: 2,
                name: 'Tops',
                url_path: 'tops',
                url_suffix: '.html',
                isActive: false,
                children: []
            }
        ],
        rootCategoryName: 'Clothing',
        firstLevelCategoryName: 'Women'
    };
    test('it renders correctly', () => {
        const instance = createTestInstance(
            <MemoryRouter>
                <Submenu {...props} />
            </MemoryRouter>
        );

        expect(instance.toJSON()).toMatchSnapshot();
    });
});
