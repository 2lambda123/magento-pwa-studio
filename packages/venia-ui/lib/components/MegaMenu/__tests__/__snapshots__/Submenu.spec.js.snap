// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Submenu renders correctly it renders correctly 1`] = `
<div
  className="submenu"
>
  <div
    className="submenuItems"
    style={
      Object {
        "minWidth": NaN,
      }
    }
  >
    <div
      className="submenuColumn"
    >
      <a
        className="link"
        href="/bottoms.html"
        onClick={[Function]}
      >
        <h3
          className="heading"
        >
          Bottoms
        </h3>
      </a>
      <ul
        className="submenuChild"
      >
        <li
          className="submenuChildItem"
        >
          <a
            className="link"
            href="/pants.html"
            onClick={[Function]}
          >
            Pants
          </a>
        </li>
      </ul>
    </div>
    <div
      className="submenuColumn"
    >
      <a
        className="link"
        href="/tops.html"
        onClick={[Function]}
      >
        <h3
          className="heading"
        >
          Tops
        </h3>
      </a>
    </div>
  </div>
</div>
`;
